### ISSUE TITLE
How can I filter by route table ?

### ISSUE THREAD

**Issue by GreatFireWall** [basil00/Divert#113]
Cause the limit of filter string, how can I filter by a batch IP range as the following route table (thousands total):
```
1.0.0.0/15
1.2.0.0/16
1.4.0.0/14
1.9.0.0/16
1.10.0.0/15
1.16.0.0/13
1.32.0.0/13
1.40.0.0/14
1.44.0.0/16
1.46.0.0/15
1.52.0.0/14
1.64.0.0/14
1.72.0.0/13
1.96.0.0/12
1.112.0.0/14
...
...
223.176.0.0/12
223.194.0.0/15
223.196.0.0/15
223.200.0.0/16
223.204.0.0/14
223.216.0.0/14
223.222.0.0/15
223.224.0.0/12
223.252.0.0/17
223.253.0.0/16
223.255.128.0/17
```

**Comment by basil00** [comment]
The standard approach is to open a WinDivert handle with the `true` filter, then do the "real" filtering in the user-mode application.  If a packet does not match any IP range, then simply re-inject it unmodified.

**Comment by GreatFireWall** [comment]
but will this cause unnecessary performance lost due to kernel-mode -> user-mode -> kernel-mode ?

**Comment by basil00** [comment]
Yes, but encoding a long list into the filter language is also inefficient (which is also the motivation for the filter language length restrictions).

Ideally the filter language should support a set inclusion operation, which would be implemented using an efficient binary search inside the driver.  However, this feature does not exist (yet).

**Comment by GreatFireWall** [comment]
understand. thanks your guide. :)
